Rocky.class.nut

```diff
@@ -540,12 +620,20 @@ class Rocky.Context {
 
     // Closes ALL contexts
     static function sendToAll(statuscode, response, headers = {}) {
-        // Send to all active contexts
-        foreach (context in _contexts) {
+        local contextsArray = [];
+
+        // Convert table into array because when contexts are removed
```

Comment:
what?!! why?

How could elements be removed? Squirrel is natively single-threaded language. You are not removing them from the loop.

Reply:
In the loop the send method is being called:
`contextsArray[i].send(statuscode, response);`
That method is what is deleting elements:
`delete Rocky.Context._contexts[id];`

The send() method is synchronous so the elements are deleted while I'm looping.

Reply:
oh damn! I feel like there is something architecturally wrong here. I would refactor it a bit to avoid copying: extract the portion that does just send (`_doSend`) and reuse it for `sendToAll` and `send`.
`sendToAll()` will do `_contexts.clear()` at the end and `send` will remove individual contexts.
does it make sense?